engine/common/cvar.c

```diff
@@ -887,6 +998,40 @@ void Cvar_Toggle_f( void )
 	Cvar_Set( Cmd_Argv( 1 ), va( "%i", v ));
 }
 
+/*
+============
+Cvar_Set_f
+
+Allows setting and defining of arbitrary cvars from console, even if they
+weren't declared in C code.
+============
+*/
+void Cvar_Set_f( void )
+{
+	int	i, c, l = 0, len;
+	char	combined[MAX_CMD_TOKENS];
+
+	c = Cmd_Argc();
+	if( c < 3 )
+	{
+		Msg( "Usage: set <variable> <value>\n" );
+		return;
+	}
+	combined[0] = 0;
+
+	for( i = 2; i < c; i++ )
```

Comment:
"set" is very powerful command in this implementation, since it ignores where the command came from.

If it came from network, it should NOT change the value of the filterable(if cl_filterstuffcmd is enabled) and privileged cvars.

If it's local, then it should be able to change any cvar (except readonly, init, etc).

Reply:
How to determine, is set local or came from network?

Reply:
There is a function in Cmd_ namespace. I don't remember how it's called but you'll figure it out.

Reply:
Is it Cmd_CurrentCommandIsPrivileged?

Reply:
I added this check